### PR TITLE
Make docker image status more granular with the percentage done (2)

### PR DESCRIPTION
**Pull request by adiprerepa**
Based off of https://github.com/codalab/codalab-worksheets/pull/3385

fixes the issue found by the stress test: https://worksheets-dev.codalab.org/bundles/0x25eb6936b59e463c95cf884caa0e4f7f

the error was:

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/codalab/worker/docker_image_manager.py", line 253, in download
    / line['progressDetail']['total']
KeyError: 'total'
```

This checks for the key `total`

### CODE REVIEW

File: codalab/worker/docker_image_manager.py [hunk 1]
```diff
@@ -246,7 +246,10 @@ def download():
                         for line in self._docker.api.pull(image_spec, stream=True, decode=True):
                             self._downloading[image_spec]['status'] = ''
                             # Set the status to the percent completed
-                            if line['status'] == 'Downloading':
+                            if (
+                                line['status'] == 'Downloading'
+                                and 'total' in line['progressDetail']
```
Comment (epicfaace) [inline codalab/worker/docker_image_manager.py:249]: We also need to ensure that `'current'` is in `line['progressDetail']`

**Review by epicfaace — COMMENTED** [review]
Looks good -- can you explain a bit how this error would have led to the `'message'` error message that showed up on the stress test result?

**Review by epicfaace — APPROVED** [review]

**Review by epicfaace — APPROVED** [review]

**Review by epicfaace — CHANGES_REQUESTED** [review]
Can you make the changes I requested?

**Comment by epicfaace** [comment]
> can you explain a bit how this error would have led to the `'message'` error message that showed up on the stress test result?

@adiprerepa can you explain a bit about what you think regarding this question?

**Comment by adiprerepa** [comment]
> > can you explain a bit how this error would have led to the `'message'` error message that showed up on the stress test result?
> 
> @adiprerepa can you explain a bit about what you think regarding this question?

Not really sure. I think it comes from https://github.com/codalab/codalab-worksheets/blob/master/codalab/worker/docker_image_manager.py#L261, and we access `_downloading[image_spec]`, which returns 'message', as opposed to accessing `_downloading[image_spec]['message']`. I have to look into it for where it's actually called, though.

Does this issue block this PR, though? Can we merge this one?

**Comment by epicfaace** [comment]
0. Add the changes from https://github.com/codalab/codalab-worksheets/pull/3385 into this PR
1. Make sure GHA builds the docker images
2. Deploy this branch to dev (venv/bin/...) (set the codalab_version to the branch name - fix-docker-image-download)
3. Run stress tests

**Comment by adiprerepa** [comment]
OK, deployed to dev, running stress tests now

**Comment by adiprerepa** [comment]
stress tests didnt pass, but interestingly, they didnt pass even before I deployed `fix-docker-image-download`
Before deploy: https://github.com/codalab/codalab-worksheets/actions/runs/746356708
After deploy: https://github.com/codalab/codalab-worksheets/actions/runs/747138730
Same error. I wonder if it's something else...

**Comment by adiprerepa** [comment]
Seems to fail because of a `401 Unauthorized` which is weird because this doesn't touch anything near the `cl wait` endpoint.

Error:
```
>> cl wait 0x509b80d1e59d4a6da5eb30fce28a1aec
WARNING:retry.api:HTTP Error 401: Unauthorized, retrying in 1 seconds...
```

**Comment by adiprerepa** [comment]
@epicfaace thoughts?

**Comment by adiprerepa** [comment]
stress tests arent passing becuase dev is down, blocked...

**Comment by adiprerepa** [comment]
https://github.com/codalab/codalab-worksheets/actions/runs/798463255

@epicfaace stress tests pass, can we merge this?

**Review by epicfaace — APPROVED** [review]

**Pull request merged**